python/ray/ml/config.py

```diff
@@ -105,6 +115,47 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         bundles = trainer_bundle + worker_bundles
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
+    @classmethod
```

Comment:
Can we make this into a regular function instead of a class method?

We are not using any class attributes anywhere, and we are planning to make ScalingConfigDataclass a user facing public API eventually, so I don't think we should expose this method for users to call themselves.

Also, the validation only makes sense in the context of Trainers for now, so I'm not sure if it should be a method on the dataclass itself.